crates/zksync/core/src/vm/env.rs

```diff
@@ -7,15 +7,50 @@ use zksync_multivm::{
 use zksync_state::interface::{ReadStorage, StoragePtr};
 use zksync_types::{
     block::{unpack_block_info, L2BlockHasher},
-    fee_model::L1PeggedBatchFeeModelInput,
+    fee_model::PubdataIndependentBatchFeeModelInput,
     StorageKey, SYSTEM_CONTEXT_ADDRESS, SYSTEM_CONTEXT_BLOCK_INFO_POSITION,
 };
 use zksync_utils::h256_to_u256;
 
+#[derive(Debug, Clone)]
+/// Values related to the era vm environment
+pub struct ZkEnv {
+    /// l1 gas price
+    pub l1_gas_price: u64,
+    /// fair l2 gas price
+    pub fair_l2_gas_price: u64,
+    /// fair pubdata price
+    pub fair_pubdata_price: u64,
+}
+
+impl Default for ZkEnv {
+    fn default() -> Self {
+        // TODO: fair pubdata price of 0 yields division by 0 error somewhere in
+        // some cases. Should investigate this edge case further
+        Self { l1_gas_price: 0, fair_l2_gas_price: 0, fair_pubdata_price: 1000 }
+    }
+}
+
+impl ZkEnv {
+    /// Compute gas per pubdata
+    pub fn gas_per_pubdata(&self) -> u64 {
+        // https://github.com/matter-labs/era-contracts/blob/aafee035db892689df3f7afe4b89fd6467a39313/system-contracts/bootloader/bootloader.yul#L86
+        let max_l2_gas_per_pubdata: u64 = 50000;
```

Comment:
I will have to check this, as the docs references differently. See [here](https://matter-labs.github.io/zksync-era/core/latest/guides/advanced/07_fee_model.html#l1-vs-l2-pricing).

Reply:
Would be nice to validate. FWIW I get the same value that's computed there when running the transactions and looking at the traces. Math comes from here: https://github.com/matter-labs/era-contracts/blob/aafee035db892689df3f7afe4b89fd6467a39313/system-contracts/bootloader/bootloader.yul#L63

Reply:
Confirmed with protocol that the docs are out of date there. 50000 is the correct value.